tx/signer_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2024 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package tx
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/secp256k1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSign(t *testing.T) {
+	// Generate a new private key for testing
+	pk, err := crypto.GenerateKey()
+	assert.NoError(t, err)
+
+	tx := new(Builder).Build()
+
+	// Sign the transaction
+	signedTx, err := Sign(tx, pk)
+	assert.NoError(t, err)
+
+	// Verify the transaction was signed
+	assert.NotNil(t, signedTx)
```

Comment:
Minor, check `tx.Origin()` to equal address derived from `pk`?

Reply:
Added !